src/v2/style-guide/index.md

```diff
@@ -0,0 +1,1805 @@
+
+1. train your brain to more easily parse most of the community code you encounter
+2. be able to copy and paste most community code examples without modification
+2. often find new hires are already accustomed to your preferred coding style, at least in regards to Vue
+
+### Priority D: Use with Caution
+
+Some features of Vue exist to accommodate rare edge cases or smoother migrations from a legacy code base. When overused however, they can make your code more difficult to maintain or even become a source of bugs. These rules shine a light on potentially risky features, describing when and why they should be avoided.
+
+[![goto from xkcd.com](https://imgs.xkcd.com/comics/goto.png)](https://xkcd.com/292/)
+
+
+
+## Priority A Rules: Essential (Error Prevention)
+
+
+
+### Multi-word component names <sup data-p="a">essential</sup>
+
+**Component names should always be multi-word, except for root `App` components.**
+
+This prevents conflicts with existing and future HTML elements, since all HTML elements are a single word.
+
+{% raw %}<div class="style-example example-bad">{% endraw %}
+#### Bad
+
+``` js
+Vue.component('todo', {
+  // ...
+})
+```
+
+``` js
+export default {
+  name: 'Todo',
+  // ...
+}
+```
+{% raw %}</div>{% endraw %}
+
+{% raw %}<div class="style-example example-good">{% endraw %}
+#### Good
+
+``` js
+Vue.component('todo-item', {
+  // ...
+})
+```
+
+``` js
+export default {
+  name: 'TodoItem',
+  // ...
+}
+```
+{% raw %}</div>{% endraw %}
+
+
+
+### Component data <sup data-p="a">essential</sup>
+
+**Component `data` must be a function.**
+
+When using the `data` property on a component (i.e. anywhere except on `new Vue`), the value must be a function that returns an object.
+
+{% raw %}
+<details>
+<summary>
+  <h4>Detailed Explanation</h4>
+</summary>
+{% endraw %}
+
+When the value of `data` is an object, it's shared across all instances of a component. Imagine, for example, a `TodoList` component with this data:
+
+``` js
+data: {
+  listTitle: '',
+  todos: []
+}
+```
+
+We might want to reuse this component, allowing users to maintain multiple lists (e.g. for shopping, wishlists, daily chores, etc). There's a problem though. Since every instance of the component references the same data object, changing the title of one list will also change the title of every other list. The same is true for adding/editing/deleting a todo.
+
+Instead, we want each component instance to only manage its own data. For that to happen,each instance must generate a unique data object. In JavaScript, this can be accomplished by returning the object in a function:
```

Comment:
Missing a space between `happen,` and `each`.

Reply:
Thanks! Fixing.